src/extensions/speech_extension/commands/parsers/transcribe_command_parser.cs

```diff
@@ -0,0 +1,65 @@
+
+namespace Azure.AI.Details.Common.CLI
+{
+    public class TranscribeCommandParser : CommandParser
+    {
+        public static bool ParseCommand(INamedValueTokens tokens, ICommandValues values)
+        {
+            return ParseCommand("transcribe", transcribeCommandParsers, tokens, values);
+        }
+
+        public static bool ParseCommandValues(INamedValueTokens tokens, ICommandValues values)
+        {
+            return ParseCommandValues("transcribe", transcribeCommandParsers, tokens, values);
+        }
+
+        public static IEnumerable<INamedValueTokenParser> GetCommandParsers()
+        {
+            return transcribeCommandParsers;
+        }
+
+        #region private data
+
+        private static INamedValueTokenParser[] transcribeCommandParsers = {
+
+            new RequiredValidValueNamedValueTokenParser(null, "x.command", "11", "transcribe;speech.transcribe"),
+
+            new ExpectOutputTokenParser(),
+            new DiagnosticLogTokenParser(),
+            new CommonNamedValueTokenParsers(),
+
+            new ExpandFileNameNamedValueTokenParser(),
+
+            new SpeechConfigServiceConnectionTokenParser(),
+            new TrueFalseNamedValueTokenParser("service.config.content.logging.enabled", "00011;00110"),
+
+            new Any1ValueNamedValueTokenParser("--languages", "source.language.config", "100;010"),
+            new Any1ValueNamedValueTokenParser("--channels", "audio.input.channels", "010"),
```

Comment:
>010 [](http://example.com/codeflow?start=86&length=3)

Does this mean --channels 2 and --input 2 mean the same thing?

Reply:
Yes... maybe it shouldn't be allowed? I can flip it to "011" to force it to be "--input-channels"

Reply:
--input isn't something I'd expect was going to mean channels.

Reply:
updated... (will be in next push)